server/data_access_layer/daos/invoice_affect.dao.js

```diff
@@ -70,4 +72,78 @@ exports.getInvoicesByDateAndEmployee = async (startDate, endDate, clientList, db
             reject(err);
         }
     });
+}
+
+
+exports.getInvoicesByDateAndCountry = async (startDate, endDate, countryCode, db = database) => {
+    return new Promise(async (resolve, reject) => {
+        try{
+            const data = await db.query(
+                "Select IH.INVOCIE_DATE, IH.ACTOR_ID, BIA.AFFECT_AMOUNT, C.COUNTRY_LABEL \
+                from [Patricia reduction].[dbo].[INVOICE_HEADER] IH, [Patricia reduction].[dbo].[BOSCO_INVOICE_AFFECT] BIA, [Bosco reduction].[dbo].[COUNTRY] C  \
+                Where IH.INVOICE_PREVIEW=0 \
+                AND IH.INVOICE_TYPE in (1,4) \
+                AND IH.INVOCIE_DATE between ? AND ? \
+                AND BIA.INVOICE_ID=IH.INVOICE_ID \
+                AND BIA.AFFECT_ACCOUNT like '%1200%' \
+                AND C.COUNTRY_LABEL = ?",
+                {
+                    replacements: [startDate, endDate, countryCode],
```

Comment:
here you are comparing c.country_label to the `countryCode` variable. firstly, this doesn't filter by country. it just adds a column called Country Label with the string that you pass in `countryCode`. if you try the query manually on the database, you will see that you always get the same amount of rows but with an additional column with the name of the country you pass. Secondly, if you're actually passing the country label, then change `countryCode` to `countryLabel`

Reply:
I disagree about what you said, if you actually test the query in the database with and without the country label, you'll see that it will filter all the invoices based on the country, it doesn't just add a new row.

Without the criteria
![proof2](https://user-images.githubusercontent.com/89820626/153727036-d2926e67-2807-48a7-bb0c-e8007e0aceb3.PNG)

With the criteria
![proof](https://user-images.githubusercontent.com/89820626/153727040-fd077df4-7c82-4fa1-9dbc-1ef4a9e1c8e6.PNG)

I'll change the variable name

Reply:
Trythe second query (with the criteria) with 'Brazil' instead of Canada. You will see what I mean by it not filtering. Also when you dont put the criteria, you get over 4 million rows which would mean that Benopit&Cote has made over 4 million invoices over 2 years (not very likely). So basically without the criteria, you are just adding a column with the name of each country in the country table and the rows you're getting are a combination of each invoice with each country label. After that, with the criteria, you're just adding a new column with the country label that you specify and the rows you're getting are just all the invoices from that time period but with the additional column that doesn't do any filtering.

Reply:
Ok I see, yeah so we need to compare the actor id of invoice header with the actor id found in the name table and then use the country code to actually filter it

Reply:
Oh nevermind there it's based on one more table in between the name_connection.

Reply:
Ok, this should work technically. Let me know what you think. It should be accurate since this query display the invoice of every employee.

![like this](https://user-images.githubusercontent.com/89820626/153727815-ba085edd-739f-47f0-a0c8-2484d3511f7c.PNG)

![like this2](https://user-images.githubusercontent.com/89820626/153727876-37915027-cddb-48d9-a887-b45436740c2e.PNG)